code/ATMOSPHERICS/components/unary/cap.dm

```diff
@@ -125,3 +125,61 @@
 		return // Coloring pipes.
 
 	return ..()
+
+/obj/machinery/atmospherics/unary/cap/bluespace
+	name = "bluespace pipe endcap"
+	desc = "A bluespace endcap for pipes."
+	icon = 'icons/obj/pipes.dmi'
+	icon_state = "bscap"
+	can_be_coloured = 0
+	var/global/list/obj/machinery/atmospherics/bspipe_list = list()
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/blue
+	color=PIPE_COLOR_BLUE
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/cyan
+	color=PIPE_COLOR_CYAN
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/green
+	color=PIPE_COLOR_GREEN
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/pink
+	color="#ff66cc"
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/purple
+	color=PIPE_COLOR_PURPLE
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/red
+	color=PIPE_COLOR_RED
+	
+/obj/machinery/atmospherics/unary/cap/bluespace/orange
+	color=PIPE_COLOR_ORANGE
```

Comment:
It's much more preferable to just limit it to /obj/machinery/atmospherics/unary/cap/bluespace and have a default color new caps will use unless they're changed. I don't see any benefit in making these unique items since you already distinguish between them via the color var.

Reply:
Well, each one connects to same-color caps, right? Seems useful to use a subtype for that instead of checking the color var, because color can be changed by other happenings.

Reply:
Ah, the code currently checks for the `color` var. I think that's a bad idea that might lead to jank if, say, Colorful Reagent shows up, or a wizard uses Color Change, or if something else happens like an admin

Reply:
Best approach would be to just have an ID var and match that. You could generate a color procedurally based on the ID if you wanted to provide visual feedback.